src/bindings/python/tests/test_graph/test_manager.py

```diff
@@ -153,39 +150,16 @@ def test_serialize_results(prepare_ir_paths):
     core = Core()
     node_constant = ops.constant(np.array([[0.0, 0.1, -0.1], [-2.5, 2.5, 3.0]], dtype=np.float32))
     node_ceil = ops.ceiling(node_constant)
-    func = Model(node_ceil, [], "Model")
+    model = Model(node_ceil, [], "Model")
 
     xml_path, bin_path = prepare_ir_paths
     pass_manager = Manager()
     pass_manager.register_pass(Serialize(path_to_xml=xml_path, path_to_bin=bin_path))
-    pass_manager.run_passes(func)
-
-    res_model = core.read_model(model=xml_path, weights=bin_path)
-    const = func.get_results()[0].input(0).get_source_output().get_node()
-    new_const = res_model.get_results()[0].input(0).get_source_output().get_node()
-
-    assert const == new_const
-
-    import gc
-    gc.collect()    # CVS-106805
-
-
-@pytest.mark.xfail(reason="CVS-106805. Delete test after fix")
-def test_serialize_results_fails(prepare_ir_paths):
-    core = Core()
-    node_constant = ops.constant(np.array([[0.0, 0.1, -0.1], [-2.5, 2.5, 3.0]], dtype=np.float32))
-    func = Model(node_constant, [], "Model")
-
-    xml_path, bin_path = prepare_ir_paths
-    pass_manager = Manager()
-    pass_manager.register_pass(Serialize(path_to_xml=xml_path, path_to_bin=bin_path))
-    pass_manager.run_passes(func)
+    pass_manager.run_passes(model)
 
     res_model = core.read_model(model=xml_path, weights=bin_path)
-    const = func.get_results()
-    new_const = res_model.get_results()
 
-    assert const == new_const
+    assert compare_functions(model, res_model)[0] is True
```

Comment:
If use of `==` isn't correct and user should apply `compare_functions()`, may we document it? From my perspective, it isn't too clear what should I use if it works for `Parameter`, but not for `Result`

Reply:
Requested rereview. Now solution is purely based on existing public API so additional module is not required to be included in installation configuration/package.